modules/rgbd/test/test_tsdf.cpp

```diff
@@ -288,230 +321,202 @@ void normalsCheck(Mat normals)
-        _params->truncateThreshold, _params->volumeDims);
-    volume->integrate(depth, _params->depthFactor, poses[0].matrix, _params->intr);
 
-    volume->raycast(poses[0].matrix, _params->intr, _params->frameSize, _points, _normals);
+    settings.volume->integrate(depth, settings.params->depthFactor, settings.poses[0].matrix, settings.params->intr);
+    settings.volume->raycast(settings.poses[0].matrix, settings.params->intr, settings.params->frameSize, _points, _normals);
     normals = _normals.getMat(af);
     points = _points.getMat(af);
     patchNaNs(points);
     anfas = counterOfValid(points);
 
     if (display)
-    {
-        imshow("depth", depth * (1.f / _params->depthFactor / 4.f));
-        renderPointsNormals(points, normals, image, _params->lightPose);
-        imshow("render", image);
-        waitKey(2000);
-    }
-
-    volume->raycast(poses[17].matrix, _params->intr, _params->frameSize, _newPoints, _newNormals);
+        displayImage(depth, points, normals, settings.params->depthFactor, settings.params->lightPose);
 
+    settings.volume->raycast(settings.poses[17].matrix, settings.params->intr, settings.params->frameSize, _newPoints, _newNormals);
     normals = _newNormals.getMat(af);
     points = _newPoints.getMat(af);
     patchNaNs(points);
     profile = counterOfValid(points);
 
     if (display)
-    {
-        imshow("depth", depth * (1.f / _params->depthFactor / 4.f));
-        renderPointsNormals(points, normals, image, _params->lightPose);
-        imshow("render", image);
-        waitKey(2000);
-    }
+        displayImage(depth, points, normals, settings.params->depthFactor, settings.params->lightPose);
+
+    // TODO: why profile == 2*anfas ?
+    float percentValidity = float(anfas) / float(profile);
 
-    float percentValidity;
-    if (profile == 0)    percentValidity = -0.5;
-    else if (anfas == 0) percentValidity = 0;
-    else                 percentValidity = float(profile) / float(anfas);
-    ASSERT_LT(0.5 - percentValidity, 0.3);
+    ASSERT_NE(profile, 0) << "There is no points in profile";
+    ASSERT_NE(anfas, 0) << "There is no points in anfas";
+    ASSERT_LT(abs(0.5 - percentValidity), 0.3) << "percentValidity out of [0.3; 0.7] (percentValidity=" << percentValidity << ")";
 }
 
-TEST(TSDF, raycast_normals)
+#ifndef HAVE_OPENCL
+TEST(TSDF, raycast_normals) { normal_test(false, true, false, false); }
+TEST(TSDF, fetch_points_normals) { normal_test(false, false, true, false); }
+TEST(TSDF, fetch_normals) { normal_test(false, false, false, true); }
+TEST(TSDF, valid_points) { valid_points_test(false); }
+
+TEST(HashTSDF, raycast_normals) { normal_test(true, true, false, false); }
+TEST(HashTSDF, fetch_points_normals) { normal_test(true, false, true, false); }
+TEST(HashTSDF, fetch_normals) { normal_test(true, false, false, true); }
+TEST(HashTSDF, valid_points) { valid_points_test(true); }
+#else
+TEST(TSDF_CPU, raycast_normals)
```

Comment:
Please remove OCL tests from this file, we have them in `ocl/` folder

Reply:
No, if I remove them there would be two equal GPU tests